src/ekat/ekat_pack_kokkos.hpp

```diff
@@ -280,8 +288,9 @@ void host_to_device(const Kokkos::Array<typename ViewT::value_type::scalar const
 {
   using PackT = typename ViewT::value_type;
   using ScalarT = typename PackT::scalar;
+  using VectorT = typename HTDVectorT<ScalarT>::type;
 
-  std::vector<ScalarT> tdata;
+  std::vector<VectorT> tdata;
```

Comment:
I'm reading that `The value of sizeof(bool) is implementation defined and might differ from 1`. I'm not sure if this ever happens on our target machines, but better be safe.

You might want to use

```
using type = char[sizeof(bool)/sizeof(char)];
```
That way you avoid risks when std::vector allocates the space.

Also, if you do this in the generic impl, then you probably don't need specializations for VectorT.

Reply:
Good catch. For now I'll add a static assert on the sizeof.